process/coordinator/processProposal.go

```diff
@@ -106,14 +108,33 @@ func (tc *transactionCoordinator) CreateMbsCrossShardDstMe(
 			"type", miniBlock.Type,
 			"round", miniBlockInfo.Round,
 			"num txs", len(miniBlock.TxHashes),
-			"total gas provided", tc.gasHandler.TotalGasProvided(), // todo: don't use the same instance used in processing, as it will be accessed in parallel
 		)
 
 		miniBlocksAndHashes = append(miniBlocksAndHashes, block.MiniblockAndHash{
 			Miniblock: miniBlock,
 			Hash:      miniBlockInfo.Hash,
 		})
 		numTransactions += uint32(len(miniBlock.TxHashes))
+
+		txsForMbs[string(miniBlockInfo.Hash)] = existingTxsForMb
+		mbsSlice = append(mbsSlice, &block.MiniBlockHeader{
+			Hash:            miniBlockInfo.Hash,
+			SenderShardID:   miniBlock.SenderShardID,
+			ReceiverShardID: miniBlock.ReceiverShardID,
+			TxCount:         uint32(len(miniBlock.TxHashes)),
+			Type:            miniBlock.Type,
+			Reserved:        miniBlock.Reserved,
+		})
+	}
+
+	lastMBIndex, _, err := tc.gasComputation.CheckIncomingMiniBlocks(mbsSlice, txsForMbs)
```

Comment:
This is called here just for one MetaBlock miniblocks, so it will (potentially) be called afterwards for the next MetaBlock if lastMBIndex == len(mbsSlice). Can the `gasComputation.CheckIncomingMiniBlocks` handle additive requests (continue to add from the new set of miniblocks to its ongoing verification) ? I don't think it supports it right now

Reply:
So I see two options:
1. Implement additive handling
2. move the handling on CreateMbsCrossShardDstMe caller

I think 1. would be simpler.

Reply:
pushed suggestion 1